src/huggingface_hub/file_download.py

```diff
@@ -48,6 +48,32 @@
 except importlib_metadata.PackageNotFoundError:
     pass
 
+_pydot_available = False
+
+try:
+    _pydot_version = importlib_metadata.version("pydot")
+    _pydot_available = True
+except importlib_metadata.PackageNotFoundError:
+    pass
+
+
+def is_pydot_available():
+    return _pydot_available
+
+
+_graphviz_available = False
+
+try:
+    _graphviz_version = importlib_metadata.version("graphviz")
+    _graphviz_available = True
+except importlib_metadata.PackageNotFoundError:
+    pass
+
+
+def is_graphviz_available():
+    return _graphviz_available
+
```

Comment:
Answering to your comment @osanseviero, I think it's time to have a `file_utils.py` module in which all of this should go.

Reply:
Doesn't have to be this PR, but would love a follow-up PR that moves this alongside other version checks to a dedicated module